sklearn/metrics/tests/test_score_objects.py

```diff
@@ -493,31 +494,75 @@ def test_scorer_sample_weight():
     estimator = _make_estimators(X_train, y_train, y_ml_train)
 
     for name, scorer in SCORERS.items():
-        if name in MULTILABEL_ONLY_SCORERS:
-            target = y_ml_test
-        else:
-            target = y_test
-        if name in REQUIRE_POSITIVE_Y_SCORERS:
-            target = _require_positive_y(target)
-        try:
-            weighted = scorer(estimator[name], X_test, target,
-                              sample_weight=sample_weight)
-            ignored = scorer(estimator[name], X_test[10:], target[10:])
-            unweighted = scorer(estimator[name], X_test, target)
-            assert weighted != unweighted, (
-                "scorer {0} behaves identically when "
-                "called with sample weights: {1} vs "
-                "{2}".format(name, weighted, unweighted))
-            assert_almost_equal(weighted, ignored,
-                                err_msg="scorer {0} behaves differently when "
-                                "ignoring samples and setting sample_weight to"
-                                " 0: {1} vs {2}".format(name, weighted,
+        if name not in REGRESSION_SCORERS:
```

Comment:
Wasn't sure how to do this outside of the for loop without copying the `SCORER` dict (@glemaitre)

Reply:
I think that this is fine. Another way could have been

```python
if name in CLF_SCORERS:
```

Reply:
Annoyingly `CLF_SCORERS` does not overlap with `CLUSTER_SCORERS` and `MULTILABEL_ONLY_SCORERS`. 

(though `REGRESSION_SCORERS` overlaps with `REQUIRE_POSITIVE_Y_SCORERS` so I could adjust the classification test)

Reply:
Ah ok so it is good.

I might have done the following to avoid a level of indent and put a nice comment

```python
if name in REGRESSION_SCORERS:
    # skip the regression scores since we evaluate the classification scores
    continue